locallib.php

```diff
@@ -799,3 +799,74 @@ function zoom_get_alternative_host_array_from_string($alternativehoststring) {
     $alternativehostarray = array_filter(explode(',', $alternativehoststring));
     return $alternativehostarray;
 }
+
+/**
+ * Get all custom user profile fields of type text
+ *
+ * @return array list of user profile fields
+ */
+function zoom_get_user_profile_fields() {
+    global $DB;
+
+    $userfields = [];
+    $records = $DB->get_records('user_info_field', ['datatype' => 'text']);
+    foreach ($records as $record) {
+        $userfields[$record->shortname] = $record->name;
+    }
+
+    return $userfields;
+}
+
+/**
+ * Get all valid options for API Identifier field
+ *
+ * @return array list of all valid options
+ */
+function zoom_get_api_identifier_fields() {
+    $options = [
+        'email' => get_string('email'),
+        'username' => get_string('username'),
+        'idnumber' => get_string('idnumber'),
+    ];
+
+    $userfields = zoom_get_user_profile_fields();
+    if (!empty($userfields)) {
+        $options = array_merge($options, $userfields);
```

Comment:
We probably want to prioritize the core fields and not use any duplication that might exist in the profile fields.
```suggestion
        $options += $userfields;
```

Reply:
done, I had the `+=` to start with....but went with `array_merge` - changed it back to `+=` now